src/client/datascience/kernel-launcher/localPythonAndRelatedNonPythonKernelSpecFinder.ts

```diff
@@ -107,8 +107,12 @@ export class LocalPythonAndRelatedNonPythonKernelSpecFinder extends LocalKernelS
             this.listGlobalPythonKernelSpecs(true, cancelToken)
         ]);
 
-        const globalPythonKernelSpecsRegisteredByUs = globalKernelSpecs.filter((item) =>
-            isKernelRegisteredByUs(item.kernelSpec)
+        // Only return global location kernelspecs if they are the old ones registered by us, or if they
+        // do not match the default kernel spec name.
+        const nonDefaultGlobalPythonKernelSpecs = globalKernelSpecs.filter(
+            (item) =>
+                isKernelRegisteredByUs(item.kernelSpec) ||
+                !item.kernelSpec.name.toLowerCase().match(isDefaultPythonKernelSpecName)
```

Comment:
Note, this isn't right (we are already removing default below). More discussion in the linked issue.